### DISCUSSION TITLE
can I updated a field only if the field request was exists?

### DISCUSSION THREAD

**Discussion by suryaadtmaja** [prisma/prisma#9291] [Q&A] [answered]
Hi,
I recently found out that if I have this model 

```prisma
model Branch {
  id       Int    @id @default(autoincrement())
  location String
  Workers Workers? @relation(fields: [workersId], references: [id])
  workersId Int?
}
```

and if I'm trying to update it only the workersId 

```ts
prisma.update.branch({
          where: {
            id: decode.id
          },
          data: {
            workersId: req.body.workersId
          }
        })
```

 prisma will returning an error
`
"message": "Illegal arguments: undefined, number"
`

and succeed if I add the required field

```ts
prisma.update.branch({
      where: {
        id: decode.id
      },
      data: {
        location: 'scranton',
        workersId: req.body.workersId
      }
})
```

Is it normal or have I miss something while creating a model, so I have a solution to fix the problem, but I'm not sure if it is the right thing, I use something like this

```
workersId: req.body.workersId
location: req.body.location ? req.body.location : undefined
```

Thank you, I'm sorry if it's not Prisma related issue.

**Comment by ryands17** [comment, marked as answer]
It cannot reproduce this on the latest Prisma version (`3.0.2`). It seems that there's some issue in the values you're passing. Here's the code that's working fine.

[test-prisma.zip](https://github.com/prisma/prisma/files/7177271/test-prisma.zip)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
